onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -499,18 +521,26 @@ def aten_arange_start_step(
     if dtype == -1:
         result = op.Range(start, end, step)
     elif _range_supported(dtype):
-        end = op.Cast(end, to=dtype)
-        start = op.Cast(start, to=dtype)
-        step = op.Cast(step, to=dtype)
-        result = op.Range(start, end, step)
+        if dtype in [INT16.dtype, INT32.dtype]:
```

Comment:
```suggestion
        if dtype in (INT8.dtype INT16.dtype, INT32.dtype):
```

I would also use an immutable type (tuple) for immutable uses